bin/api_diff.ml

```diff
@@ -4,18 +4,15 @@ let run (`Ref_cmi reference) (`Current_cmi current) =
   let current = Cmi_format.read_cmi current in
   let reference = Cmi_format.read_cmi reference in
   let typing_env = Env.empty in
-  let coercion1 () =
-    Includemod.signatures typing_env ~mark:Mark_both reference.cmi_sign
-      current.cmi_sign
+  let coercion1 =
+    Api_watch_diff.diff_interface ~reference:reference ~current:current in
+  let coercion2 =
+    Api_watch_diff.diff_interface ~reference:current ~current:reference
   in
-  let coercion2 () =
-    Includemod.signatures typing_env ~mark:Mark_both current.cmi_sign
-      reference.cmi_sign
-  in
-  match (coercion1 (), coercion2 ()) with
-  | Tcoerce_none, Tcoerce_none -> Printf.printf "API unchanged!\n"
-  | _, _ -> Printf.printf "API changed!\n"
-  | exception Includemod.Error _ -> Printf.printf "API changed!\n"
+  match (coercion1, coercion2) with
+  | Tcoerce_none, Tcoerce_none -> false
+  | _, _ -> true
+  | exception Includemod.Error _ -> true
```

Comment:
The goal is to move all of this logic into the `diff_interface` function, then introduce a conditional here that will check the boolean result of that function to decide which output to print.

Reply:
@shonfeder I have implemented the changes you requested in the PR. Please review them.